packages/material-ui/src/styles/experimentalStyled.test.js

```diff
@@ -402,5 +402,63 @@ describe('experimentalStyled', () => {
         marginTop: '8px',
       });
     });
+
+    it('should set displayName properly', () => {
+      const Component = styled(
+        'div',
+        { shouldForwardProp: (prop) => prop !== 'variant' && prop !== 'size' && prop !== 'sx' },
+        { name: 'Component' },
+      )`
+        width: 200px;
+        height: 300px;
+      `;
+
+      expect(Component.displayName).to.equal('Component');
+    });
+
+    it('should set displayName as name + slot if both are specified', () => {
+      const Component = styled(
+        'div',
+        { shouldForwardProp: (prop) => prop !== 'variant' && prop !== 'size' && prop !== 'sx' },
+        { name: 'Component', slot: 'Root' },
+      )`
+        width: 200px;
+        height: 300px;
+      `;
+
+      expect(Component.displayName).to.equal('ComponentRoot');
+    });
+
+    it('should set the className when generating the classes', () => {
+      const Component = styled(
+        'div',
+        { shouldForwardProp: (prop) => prop !== 'variant' && prop !== 'size' && prop !== 'sx' },
+        { name: 'Component', slot: 'Slot' },
+      )`
+        width: 200px;
+        height: 300px;
+      `;
+
+      const { container } = render(<Component>Test</Component>);
+
+      expect(container.firstChild.getAttribute('class').includes('MuiComponent-slot')).to.equal(
+        true,
+      );
```

Comment:
If it should have a class with the name `MuiComponent-slot` prefer 
```suggestion
      expect(container.firstChild).to.have.class('MuiComponent-slot');
```
Otherwise we should use `classList` to find a class that matches with RegExp. Otherwise it's not clear if the class should start or end with the string.

Reply:
Will use classList and regexp, as the class looks something like `hash-MuiComponent-slot`.

Reply:
If it's prefixed by a hash why does it even matter that it contains `MuiComponent-slot`? Are people supposed to write CSS declarations like

```css
[class*="MuiComponent-slot"] {}
```
?

Reply:
The ending is supposed to be the same with the utility class, to improve readability. The component will anyway add the utility class `MuiComponent-slot`, but that is not responsibility of the `styled()` utility. This is just improving the generated class from emotion from `hash` to `hash-MuiComponent-slot`.

Reply:
The tests are updated.